cpp/arcticdb/pipeline/frame_utils.hpp

```diff
@@ -149,6 +149,8 @@ void aggregator_set_data(
                     }
                 }
             }
+        } else if constexpr (is_empty_type(dt)) {
```

Comment:
I think the `else` represents all numeric types? In which case I think it would be more explicit to have `else if constexpr (is_numeric_type(dt))`

Reply:
Are all of the none-string types numeric types, and is this a logical fact or just a coincidental thing at the moment?

Reply:
Until now, yes, coincidentally all non-numeric types were string types. @vasil-pashov please do a scan of the code to see if this assumption is made elsewhere, particularly in the filtering/projection code.

Reply:
The `bool` type is neither string nor numeric type. However up to now it was processed by the `else` statement in this snippet.

Reply:
I think what we should do in the relatively short term is to move the string handling code into the type handlers. That way we will still have two branches: 1) thing that are simple numeric types and don't require any language-specific handling like ints and floats, and 2) everything else - strings, decimals, numpy arrays, boxed Python types that can also be None etc. 

Unfortunately I think we'll need to get the existing code to read and write these types moved over from the old project more or less as-is first, because it's blocking some users, and refactoring string handling etc is much too big a job to attempt at the same time. Integrating the new object types into the data processing framework is something that we will need to do later, and I'm not sure it will even make sense in all cases. We may end up (as is the case with pickled and custom normalized data) with things that we can store and retrieve, but not sensibly do numerical operations on, at least for the time being.

Reply:
I've made a list of all function which behave as if there are only numbers and strings. But since it's out of the scope of the PR I won't change them.